lib/rspec/mocks/method_double.rb

```diff
@@ -68,6 +73,11 @@ def define_proxy_method
         end
 
         @method_is_proxied = true
+      rescue RuntimeError => e
```

Comment:
I'd prefer this to catch `FrozenError`, but sadly, it was only introduced in Ruby 2.5
It is worth adding a comment, so can we change it once we drop support for older Rubies. In RSpec 4 we're dropping support for Ruby < 2.3.